suzieq/poller/worker/coalescer_launcher.py

```diff
@@ -78,48 +87,45 @@ async def _monitor_coalescer(self):
         """This function calls _start_coalescer() and check if the process
         dies
         """
-        fd = 0
-        # Check to see file lock is possible
-        while not fd:
+
+        starting_attemps = 0
+
+        # We would like to attempt the coalescer startup self.max_attempts
+        # times
+        while starting_attemps < self.max_attempts:
             if not self.coalescer_process:
                 logger.info('Starting Coalescer')
             elif self.coalescer_process.returncode == errno.EBUSY:
                 logger.warning('Trying to start coalescer')
+            else:
+                logger.warning('Restarting the coalescer')
 
             # Try to start the coalescer process
             self.coalescer_process = await self._start_coalescer()
-
             if not self.coalescer_process:
-                os.kill(os.getpid(), signal.SIGTERM)
-                return
-
-            # Initial sleep to ensure that the coalescer starts up
-            await asyncio.sleep(10)
-            coalesce_dir = self.cfg.get('coalescer', {})\
-                .get('coalesce-directory',
-                     f'{self.cfg.get("data-directory")}/coalesced')
-
-            # In order to be sure the coalescer correctly started we need to
-            # check if it got the lock file, if so we are not able to acquire
-            # the lock and we can proceed monitoring the coalescer process,
-            # otherwise we release the lock and retry starting the process
-            fd = ensure_single_instance(f'{coalesce_dir}/.sq-coalescer.pid',
-                                        False)
-            if fd > 0:
-                # unlock and try to start process
-                try:
-                    fcntl.flock(fd, fcntl.F_UNLCK)
-                    os.close(fd)
-                except OSError:
-                    pass
-                continue
+                raise PollingError('Unable to start the coalescer')
```

Comment:
Lets not print this unless there's no coalescer and we can't start it. If there's a coalescer already running, printing this error may scare users.

Reply:
Ack, thx for clarifying.